mod_ci/controllers.py

```diff
@@ -461,6 +464,44 @@ def progress_reporter(test_id, token):
                     g.github['repository'])
                 # If status is complete, remove the Kvm entry
                 if status in [TestStatus.completed, TestStatus.canceled]:
+                    u1 = GeneralData.query.filter(
+                        GeneralData.key == 'average_time').first()
+                    average_time = 0
+                    if u1 == None:
+                        finished_tests = g.db.query(TestProgress.test_id).filter(
+                                        TestProgress.status.in_([TestStatus.canceled, TestStatus.completed])
+                                        ).subquery()
+                        times2 = g.db.query(TestProgress).filter(
+                                    and_(TestProgress.test_id.in_(finished_tests), TestProgress.status.in_(
+                                        [TestStatus.building,TestStatus.completed,TestStatus.canceled]))
+                                    ).subquery()
+                        finished_tests = times2
+                        times = g.db.query(finished_tests.c.test_id,label('time',func.group_concat(
+                                finished_tests.c.timestamp))).group_by(finished_tests.c.test_id).all()
+                        for p in times:
+                            k = p.time.split(',')
+                            pr1 = datetime.strptime(k[0], '%Y-%m-%d %H:%M:%S')
+                            pr2 = datetime.strptime(k[1], '%Y-%m-%d %H:%M:%S')
+                            sec = (pr1-pr2).total_seconds()
```

Comment:
We can safely assume that pr2 always will be later than pr1. No need to check for negative values